cloud_experts_tutorials/cloud-experts-getting-started/cloud-experts-getting-started-openshift-concepts.adoc

```diff
@@ -0,0 +1,88 @@
+
+==== Speed
+Instead of building multiple layers in a single Dockerfile, S2I encourages authors to represent an application in a single image layer. This saves time during creation and deployment and allows for better control over the output of the final image.
+
+==== Security
+Dockerfiles are run without many of the normal operational controls of containers. They usually run as root and have access to the container network. S2I can control what permissions and privileges are available to the builder image since the build is launched in a single container. In concert with platforms like OpenShift, S2I allows administrators to control what privileges developers have at build time.
+
+== Routes
+An OpenShift Route exposes a service at a hostname so that external clients can reach it by name. When a Route object is created on OpenShift, it gets picked up by the built-in HAProxy load balancer to expose the requested service and make it externally available with the given configuration. Similar to the Kubernetes Ingress object, Red Hat created the concept of Route to fill a need and then contributed the design principles behind it to the community, which heavily influenced the Ingress design. A Route does have some additional features as can be seen in the following chart.
+
+image::ostoy-routes_ingress_comparison.png[Comparing routes and ingresses]
+
+[NOTE]
+====
+DNS resolution for a hostname is handled separately from routing. Your administrator might have configured a cloud domain that will always correctly resolve to the router or modify your unrelated hostname DNS records independently to resolve to the router.
+====
+
+An individual route can override some defaults by providing specific configurations in its annotations. See xref:../../networking/routes/route-configuration.adoc#nw-route-specific-annotations_route-configuration[route specific annotations] for more details.
+
+== ImageStreams
+An ImageStream stores a mapping of tags to images, metadata overrides that are applied when images are tagged in a stream, and an optional reference to a Docker image repository on a registry.
+
+=== ImageStream benefits
+Using an ImageStream makes it easy to change a tag for a container image. Otherwise to change a tag you need to download the whole image, change it locally, then push it all back. Also promoting applications by having to do that to change the tag and then update the deployment object entails many steps. With ImageStreams you upload a container image once and then you manage its virtual tags internally in OpenShift. In one project you might use the developer tag and only change a reference to it internally, while in production you might use a production tag and also manage it internally. You do not have to deal with the registry.
```

Comment:
Various suggestions:
```suggestion
Using an image stream makes it easier to change a tag for a container image. Otherwise, to manually change a tag, you must download the image, change it locally, then push it back. Promoting applications by manually changing a tag and updating the deployment object entails many steps.

With image streams, you upload a container image once, and then you manage its virtual tags internally in OpenShift. In one project you might use a development tag and only change a reference to it internally, while in production you might use a production tag and also manage it internally. You do not have to deal with the registry.
```